New package: MinCostFlows v0.1.2

JuliaRegistrator opened the pull request:
- Registering package: MinCostFlows
- Repository: https://github.com/NREL/MinCostFlows.jl
- Created by: @GordStephen
- Version: v0.1.2
- Commit: 3e0168ff93c0d95cae96b5a1a42ca03c7576efc4
- Reviewed by: @GordStephen
- Reference: https://github.com/NREL/MinCostFlows.jl/commit/3e0168ff93c0d95cae96b5a1a42ca03c7576efc4#commitcomment-149855397
- Description: Fast min-cost flow solver for network optimization in PRAS
<!-- bf0c69308befbd3ccf2cc956ac8a46712550b79fc9bfb5e4edf8f833f05f4c18b06eddad8845b45beb9f45c2b8020dd60ad473a3339317bf4512ec43afe7a7e57a39e5776a69d5ef462c05b36c9c0ddc4aedbc8c4df39cbfdfc3e829a1a46bf4dd9807fba764b4274bff365f45239e4e0154df5d627a46e584ff255dda9e1ef03dadac6e04431c4c5694c34984fa1fc9d835d49c2b73e3f536830d9b7579e3c53c69b75733afae5a25aa319893749725a68b95c1aaaff8f0792bfc682249f02f7b3f95a7f7b681315266061b36e7c05191e291b8e00d5797e73bc2e564eb3175 -->

GordStephen commented:
Regarding the license: [there is indeed a license](https://github.com/NREL/MinCostFlows.jl/blob/main/LICENSE.md), but it's "modified" MIT Expat . My understanding is that, at least in 2019 when this software was licensed, it was corporate policy to require that "THE UNITED STATES, THE UNITED STATES DEPARTMENT OF ENERGY, OR ANY OF THEIR EMPLOYEES" be explicitly added to the liability waiver language.

Hopefully this isn't a blocker. If it is, I can talk to our tech transfer people about relicensing the software, but if there's any latitude to accept it as-is that would make things a lot easier.

[noblock]

goerz commented:
[noblock]

My understanding is that there was a decision that an OSI-approved license is a hard requirement. There is no leeway for any kind of modifications beyond what the automated bot still manages to recognize. I wasn't involved in that discussion, but I think the argument was that the Julia community is not equipped to understand what modifications to a license might legally imply. Even if it's a trivial modification such as this one, I'm not sure if there's any registry maintainer who would want to manually override the check somehow. This is especially true because AFAIK the license check is not just performed on the initial registration, but also on all subsequent updates. So, every update of the package would have to be merged manually.

Your best bet is probably to use a LICENSE file that makes the bot happy. If it's not possible to sneak in the extra text somehow (the bot does swallow some minor modifications, possibly at the beginning and end, not sure), I would include the extra text in some other place. For example, put the current modified license text at the bottom of your README. I'm not a lawyer, but I don't think the unmodified MIT and your modified MIT are in any way incompatible, or that the note would be any more or less binding if it appeared in the README. Quite honestly, my common-sense legal opinion is that the extra note about the U.S. government is completely redundant. In that vein, you _could_ try to clarify whether that extra language is still necessary. Or, probably better, don't make a fuss about it and find a pragmatic workaround. I suspect you're not going to get into any kind of trouble (even if you were to just completely drop the extra language from the LICENSE), but make your own judgement on that.

GordStephen commented:
Ok, thanks - that's understandable. I suspect this won't be a big deal to change on our end, but with the holiday here it'll just take a few days to get sorted.

JuliaRegistrator committed:
New package: MinCostFlows v0.1.2

UUID: 62286e6e-1779-56f1-888a-1c0056788ce0
Repo: https://github.com/NREL/MinCostFlows.jl.git
Tree: 8f52000f0db84f9bda8114c874410e69ce36f067

Registrator tree SHA: 17aec322677d9b81cdd6b9b9236b09a3f1374c6a
